### DISCUSSION TITLE
Difference between 2 frequencies on Status page

### DISCUSSION THREAD

**Discussion by nworbekim** [jgyates/genmon#1120] [Questions] [answered]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).

### Version

1.19.03

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

Why are there 2 different frequency values shown on the Status page (engine frequency & unsupported sensors frequency)?

One frequency is listed under "Engine" and the 2nd is listed under "Unsupported Sensors".
The Unsupported Sensors frequency displays frequencies to a tenth while the Engine frequency is whole numbers.

I see the code using different data sources in the GetFrequency function but don't understand why you're calculating the frequency twice.

**Comment by jgyates** [comment, marked as answer]
You did not include your controller type so I am going to assume you are asking about Generac Evolution / Nexus. 

"Unsupported Sensors" are not displayed by default. There are enabled by setting the "Display Unknown Sensors" option on the Settings page. Since they are not enabled by default I assume that you have enabled them on the Settings page. 

Everything displayed under "Unsupported Sensors" are values that appear to be, a sensor of some type, may not be supported on all controllers and on some controllers appear to change with the operation of the controller. These values are labeled "Unsupported" as I don't know what they do with certainty, however I give the option to display them in hopes that others may provide some input or guesses as to what these value represent. In some cases the label of the Unsupported Sensor is my best guess at what the modbus register represents. Also, the vales displayed are dependent to some degree on the type of Evolution / Nexus controller is detected.

the code for displaying unknown sensors is here:
https://github.com/jgyates/genmon/blob/c2ec011114d58394a7633240ca0b631a0dceddf3/genmonlib/generac_evolution.py#L2614

The call to GetFrequency from The  DisplayUnknownSensors calculates the Frequency based on the RPM and the Active Rotor Poles. This is listed here as it is considered experimental and I don't know how useful it would be to display this in normal operation. I thought it may be helpful in sanity checking the frequency reported from the controller or detecting if a rotor pole is disconnected from the generator.

In short, "Unsupported Sensors" are unknown or unvalidated data that is read from modbus registers or calculated from modbus registers that I don't have certainty of it's usefulness or purpose.

The frequency listed under "engine" from GetFrequency and is read directly from the modbus registers (assuming we are talking about Evolution Nexus.

**Reply by nworbekim** [reply]
Thanks for the complete answer!

Sorry about not including the controller type -- yes it is a Generac Nexus

I should have included that my  interest in the "unsupported sensors frequency" is the lack of precision in the standard  frequency value.
